document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoIdentifierPrefixExpressionParser.java

```diff
@@ -5,11 +5,9 @@
 import java.util.Optional;
 import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
 
-final class MongoIdentifierPrefixingSelectingExpressionParser
-    extends MongoSelectingExpressionParser {
+final class MongoIdentifierPrefixExpressionParser extends MongoSelectTypeExpressionParser {
```

Comment:
I thought, we agreed upon `MongoIdentifierPrefixingSelectionParser` at https://github.com/hypertrace/document-store/pull/70#discussion_r806521242, right?

`MongoIdentifierPrefixExpressionParser` feels like this class is parsing an expression type called `IdentifierPrefixExpression`

Reply:
Better to maintain the gerund form, when the parser is doing something extra on top of the base parser.

Reply:
`MongoIdentifierPrefixingParser` also works.

Reply:
yeah right, current name needs to be changed. 

so the problem with `MongoIdentifierPrefixingParser` is that it feels like two actions prefixing and parsing, if we want to maintain the gerund form should we do away with the `parser` suffix?

Reply:
Yeah. `MongoIdentifierPrefixer` sounds just about right. The only problem is with the inheritance ("is-a" relationship).

With the superclass, we are saying that `MongoIdentifierPrefixer` is a `MongoSelectTypeExpressionParser`, which doesn't sound right.

Reply:
I think, that's where decoration (and decorator-pattern) comes into picture, right.
A decorator "decorates" normal behaviour at run time.
Here, `MongoIdentifierPrefixingParser` decorates the normal `MongoIdentifierExpressionParser` at run time (by prefixing with "$").

Reply:
yeah right, this the act of parsing and prefixing makes sense from a decorate pattern perspective, suggestion taken!